### ISSUE TITLE
char input is not working in multi glfw windows

### ISSUE THREAD

**Issue by nyaoouo** [pyimgui/pyimgui#324]
reproduce code: 
```python
import OpenGL.GL as gl
import glfw
import imgui
from imgui.integrations.glfw import GlfwRenderer


def main():
    if not glfw.init():
        raise Exception("glfw can not be initialized")

    glfw.window_hint(glfw.CONTEXT_VERSION_MAJOR, 3)
    glfw.window_hint(glfw.CONTEXT_VERSION_MINOR, 3)
    glfw.window_hint(glfw.OPENGL_PROFILE, glfw.OPENGL_COMPAT_PROFILE)

    window_1 = glfw.create_window(512, 512, 'window 1', None, None)
    glfw.make_context_current(window_1)
    imgui_ctx_1 = imgui.create_context()
    imgui.set_current_context(imgui_ctx_1)
    imgui_renderer_1 = GlfwRenderer(window_1)

    window_2 = glfw.create_window(512, 512, 'window 2', None, window_1)
    glfw.make_context_current(window_2)
    imgui_ctx_2 = imgui.create_context()
    imgui.set_current_context(imgui_ctx_2)
    imgui_renderer_2 = GlfwRenderer(window_2)

    glfw.swap_interval(1)
    imgui.get_io().ini_file_name = b''

    val_1 = 'hello world 1'
    val_2 = 'hello world 2'
    while not glfw.window_should_close(window_1) and not glfw.window_should_close(window_2):
        glfw.poll_events()

        glfw.make_context_current(window_1)
        imgui.set_current_context(imgui_ctx_1)
        imgui_renderer_1.process_inputs()
        imgui.new_frame()

        imgui.set_next_window_position(5, 5, imgui.FIRST_USE_EVER)
        imgui.set_next_window_size(500, 500, imgui.FIRST_USE_EVER)
        imgui.begin('window1', True)
        _, val_1 = imgui.input_text('val', val_1, 100)

        imgui.end()
        imgui.end_frame()
        imgui.render()
        gl.glClearColor(1, 1, 1, 1)
        gl.glClear(gl.GL_COLOR_BUFFER_BIT)
        imgui_renderer_1.render(imgui.get_draw_data())
        glfw.swap_buffers(window_1)

        glfw.make_context_current(window_2)
        imgui.set_current_context(imgui_ctx_2)
        imgui_renderer_2.process_inputs()
        imgui.new_frame()

        imgui.set_next_window_position(5, 5, imgui.FIRST_USE_EVER)
        imgui.set_next_window_size(500, 500, imgui.FIRST_USE_EVER)
        imgui.begin('window2', True)
        _, val_2 = imgui.input_text('val', val_2, 100)

        imgui.end()
        imgui.end_frame()
        imgui.render()
        gl.glClearColor(1, 1, 1, 1)
        gl.glClear(gl.GL_COLOR_BUFFER_BIT)
        imgui_renderer_2.render(imgui.get_draw_data())
        glfw.swap_buffers(window_2)

    glfw.destroy_window(window_1)
    glfw.destroy_window(window_2)
    imgui.destroy_context(imgui_ctx_1)
    imgui.destroy_context(imgui_ctx_2)
    glfw.terminate()


if __name__ == '__main__':
    main()
```

in the above code, window 1 can only use backspace to remove char, but char input at window 1 is redirect to window 2

**Comment by KinoxKlark** [comment]
Thanks for reaching out!

I confirm that this occurs on version 2.0.0. I had a quick look at everything context related, and nothings seems fishy to me.

My guess is that it has to do with the `char_callback()` in `imgui/integrations/glfw.py`
```
    def char_callback(self, window, char):
        io = imgui.get_io()

        if 0 < char < 0x10000:
            io.add_input_character(char)
```

Coupled with the order of execution:
```
glfw.poll_events()
# ...
imgui.set_current_context(imgui_ctx_1)
# ...
imgui.set_current_context(imgui_ctx_2)
```

It will need some further investigation and I didn't had time to test this properly: I would guess the issue is that this callback is called once `glfw.poll_event()` is called. Thus, at the time, context hasn't switch yet and we are still using the context2 of previous frame.

What you should be able to do is to create a dict `{window: context}` and redefine the callback such that it gets the proper context when it is called.

This will have to be properly done in the various implementations. I won't have time to do it right now. If anyone is inspired, feel free :)

**Comment by KinoxKlark** [comment]
Fixed by #325
